File: Array/zhanchengguo/0617-化栈为队/Main.java
package com.hlg.daydaytobusiness;

import java.util.Stack;

public class Main {

    void main() {
        MyQueue obj = new MyQueue();
        obj.push(6);
        int param_2 = obj.pop();
        int param_3 = obj.peek();
        boolean param_4 = obj.empty();
    }
}

class MyQueue {

    private Stack<Integer> st1, st2;

    /**
     * Initialize your data structure here.
     */
    public MyQueue() {
        st1 = new Stack<>();
        st2 = new Stack<>();
    }

    /**
     * Push element x to the back of queue.
     */
    public void push(int x) {
        st1.push(x);
    }

    /**
     * Removes the element from in front of queue and returns that element.
     */
    public int pop() {
        int value = peek();
        st2.pop();
        return value;
    }

    /**
     * Get the front element.
     */
    public int peek() {
        if (st2.empty()) {
            while (!st1.empty()) {
                int value = st1.pop();
                st2.push(value);
            }
        }
        return st2.peek();
    }

    /**
     * Returns whether the queue is empty.
     */
    public boolean empty() {
        return st1.empty() && st2.empty();
    }
}

File: Array/zhanchengguo/0617-化栈为队/main.kt
package com.hlg.daydaytobusiness

import java.util.*

fun main() {
    val obj = MyQueue()
    obj.push(6)
    val param_2 = obj.pop()
    val param_3 = obj.peek()
    val param_4 = obj.empty()
}

class MyQueue {

    private val st1: Stack<Int>
    private val st2: Stack<Int>

    /**
     * Initialize your data structure here.
     */
    init {
        st1 = Stack()
        st2 = Stack()
    }

    /**
     * Push element x to the back of queue.
     */
    fun push(x: Int) {
        st1.push(x)
    }

    fun pop(): Int {
        val value = peek()
        st2.pop()
        return value
    }

    /**
     * Get the front element.
     */
    fun peek(): Int {
        if (st2.empty()) {
            while (!st1.empty()) {
                val value = st1.pop()
                st2.push(value)
            }
        }
        return st2.peek()
    }

    /**
     * Returns whether the queue is empty.
     */
    fun empty(): Boolean {
        return st1.empty() && st2.empty();
    }
}
